### PR TITLE
chore(tests): update to Intern 3

### PR DESCRIPTION
**Pull request by vladikoff**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,7 +86,7 @@
     "grunt-jsonlint": "1.0.4",
     "grunt-nsp-shrinkwrap": "0.0.3",
     "grunt-todo": "0.4.0",
-    "intern-geezer": "2.1.1",
+    "intern": "theintern/intern",
```
Comment (shane-tomlinson) [inline package.json]: YAAAAAASSSS!

File: tests/functional/change_password.js [hunk 2]
```diff
@@ -308,7 +308,7 @@ define([
           return FunctionalHelpers.getVerificationLink(email, 0);
         })
         .then(function (verificationLink) {
-          return self.get('remote').get(require.toUrl(verificationLink));
+          return self.remote.get(require.toUrl(verificationLink));
```
Comment (shane-tomlinson) [inline tests/functional/change_password.js]: After seeing this 400 or so times, part of me feels like we should have abstracted this away. That was largely my intent with https://github.com/mozilla/fxa-content-server/blob/dc048832e1db52f46621177bf98a4f203714c204/tests/functional/lib/helpers.js#L587. Perhaps as part two, or a separate PR, we could replace most of the boilerplate with a call to the helper.

**Comment by shane-tomlinson** [comment]
@vladikoff - from your POV, what needs to be done yet?

**Comment by shane-tomlinson** [comment]
:heart: Watching the tests run is so much more fun!

**Comment by vladikoff** [comment]
> @vladikoff - from your POV, what needs to be done yet?

we might be able to merge this if I lock it to some git hash and then tweak from there

**Comment by shane-tomlinson** [comment]
@vladikoff - I had to make this update for the tests to run:

```diff
diff --git a/tests/functional/email_opt_in.js b/tests/functional/email_opt_in.js
index f56f98d..cae6dad 100644
--- a/tests/functional/email_opt_in.js
+++ b/tests/functional/email_opt_in.js
@@ -169,7 +169,7 @@ define([
           return FunctionalHelpers.getVerificationLink(email, 0);
         })
         .then(function (verificationLink) {
-          return self.get('remote').get(require.toUrl(verificationLink));
+          return self.remote.get(require.toUrl(verificationLink));
         })
 
         .findByCssSelector('#fxa-settings-header')
```

**Comment by vladikoff** [comment]
@shane-tomlinson ah yes good catch! probably need to rebase

**Comment by vladikoff** [comment]
Rebase and updated to support latest functional tests.
locked to a git hash

**Comment by shane-tomlinson** [comment]
@vladikoff - looks like another rebase is needed.

**Comment by vladikoff** [comment]
@shane-tomlinson updated! :telescope:

**Comment by shane-tomlinson** [comment]
@vladikoff - uh oh. @TDA's PR collided with this.

**Comment by shane-tomlinson** [comment]
Once Travis goes green, merge it!

r=me

Nice one @vladikoff

**Pull request merged**